ssl/statem/extensions_clnt.c

```diff
@@ -663,7 +766,14 @@ static int add_key_share(SSL_CONNECTION *s, WPACKET *pkt, unsigned int curve_id)
         SSLfatal(s, SSL_AD_INTERNAL_ERROR, ERR_R_INTERNAL_ERROR);
         goto err;
     }
-
+# ifndef OPENSSL_NO_ECH
+    if (s->ext.ech.ch_depth == 1) { /* stash inner */
+        EVP_PKEY_up_ref(key_share_key);
```

Comment:
Please check the result.

Reply:
sure; that code will disappear soon anyway once we rebase as it's replaced with a function to up-ref the array of key shares needed once hybrid PQ stuff is supported (I have such code in another branch)